### ISSUE TITLE
Update scraper to work with latest xbox.com update

### ISSUE THREAD

**Issue by jasonclemons** [XboxLeaders/XboxLiveAPI#73]
With the latest xbox.com update, the scraper no longer returns the proper information. Rewrite it so that it can scrape the right information.

**Comment by jbowens** [comment]
@aayushranaut did you ever figure out RPS authentication?

**Comment by aayushranaut** [comment]
@jbowens Nah, I forgot about it after a few tries.

**Comment by thomasm789** [comment]
Is there any more updates on your progress guys?? I'm currently looking at getting gamercard information, online status etc for a crews website that I am developing.

How far did you guys get since Feb?

**Comment by dregimbal1** [comment]
Hate to kick up an old thread but in case anyone ever stumbles through here looking for a solution to this bug I wrote up my few cents on pastebin (because it is long winded haha). Without a Microsoft Partner ID you will not be able to send party invites, initiate game invites - vice versa for accepting them, or speak into parties (without proper certificates). That said, you can still get a lot of information doing it the proper way. If you are looking to provide a service you won't be able to do much without the provider of that service storing your password and email is almost-plain-text (you can encrypt it and make it hard to read but still). I have tried authorizing users on my own site using OAuth and using that access_token (but that token does not have access to the scope we are looking for so it's worthless)

http://pastebin.com/LitnfcyS

**Comment by djekl** [comment]
Just to let you know, authorising a user via oAuth isn't impossible. I have done it successfully over on https://xboxapi.com (that way you don't need to store your password with us, or even use it for a one time login via us)

**Comment by dregimbal1** [comment]
I have done that too but found the Client ID does not have access to grant permission to the scope:  service::user.auth.xboxlive.com::MBI_SSL

**Comment by djekl** [comment]
No no, I have a full Official oAuth package and login process for a user. If you wish, sign up for a free account and test it out for yourself. I FULLY authenticate the user against the Official Xbox API via oAuth

**Comment by dregimbal1** [comment]
Yeah but you're still technically 'scraping' credentials, right? If it were done through oAuth users would be given something like this:
<img width="1206" alt="screen shot 2016-05-11 at 2 49 01 pm" src="https://cloud.githubusercontent.com/assets/6709852/15192547/a9bf8062-1787-11e6-8727-70c07374e016.png">
Where they enter their credentials through Microsoft directly and you are given an access_token. That access_token is what your Client ID (website) would use to generate further tokens to get access to xbox live services.

What endpoint are you using to authenticate?

EDIT:

I see you using scope=Xboxlive.signin+Xboxlive.offline_access
I've tried that too but the access_token I got back wouldn't let me make any requests doing it the same as the manual way

**Comment by djekl** [comment]
No, I am using the actual oAuth process. If you try it then you are redirected away from XboxAPI and towards the login.live page. And I am sorry, but I am not willing to share this information.

As I have stated, its a fully functional oAuth process, redirects to `login.live.com` then over to `xbx-api.com` as `xboxapi.com` has a banned word (go figure), this then redirects back to `xboxapi.com` with the access token in the url fragment and finishes off the process there.

**Comment by markadrake** [comment]
I'm banging my brains out trying to solve this too. I've authenticated with the required credentials, but since I don't know what the endpoints are to request the profile data I can't go any further. 

@regimbal93 that writeup looks nice, how far does it get you? I'm looking to pull back the gamertag of the user & achievements. I'd also like to know if they are online, but that's not a necessity.

**Comment by dregimbal1** [comment]
A lot of the endpoints have been revealed (A few listed in this [pastebin](http://pastebin.com/LitnfcyS))

Which ones you have in mind?

The only authentication I was able to figure out was more of a jimmy rig .. It would be nice to know the way to do it via Xbox OAuth but that way doesn't seem to cough up the right scope for certain things

**Comment by markadrake** [comment]
Hi @regimbal93 and thanks for replying back so quickly. I'm looking to create an OAuth provider for ASP.NET (Core) to handle all of the work you described via pastebin. Thanks so much for your hard work  I'm sure it took a lot of time to figure this out.

You don't do any C# work do you? :) I'm essentially building a class like one found in this GitHub repo (https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers).

I could really use the help as I've got enough on my plate. After the OAuth takes place, I'm only looking to pull back the gamertag and achievements. The purpose being that I can validate they own (or at least played) a game.

I've tried to speak with people at the ID program at Xbox but between their autoresponders and carelessness to develop relationships with people and business owners I've received no answers, no introductions, nothing. So I'm just going to have to do it the hacky way it seems. 

I'm looking to do this with Sony too in the future.

**Comment by dregimbal1** [comment]
ID@Xbox is looking for studios who have a game to push to their marketplace. The "Xbox API" one day apparently will be public but in the meantime it is a thick glass wall. It quickly turns into a slippery slope. Eventually people will build bots and you will have more spam in your inbox, random friend requests, etc. 

Ideally you are a MS Partner with a Client ID that has access to the secure auth.xbox.com scope. Since for us this is not the case we are left with two scopes: Xboxlive.signin Xboxlive.offline_access both of which are open to the public but the endpoints that go with them are to be determined. I've tried reverse engineering it but I have not come up with a solution yet. 

I figure the scope we know to be open has no relation to the more complex/cool API used in the smart glass apps. Something more simple like a POST to https://live.xbox.com/en-US/Friends/List with the access_token in the header (similar calls as seen on xbox.com)

I'll take another look this weekend. I am pretty busy too but I would definitely help you where I can to tear down this glass wall. Do you use Skype?
